core/unix/os_exports.h

```diff
@@ -112,6 +115,17 @@
 # error NYI
 #endif
 
+#if defined(MACOS) && defined(X64)
+/* FIXME: current pthread_t struct has the first TLS entry at offset 28. We should
+ * provide a dynamic method to determine the first entry for forward compatability.
+ * Starting w/ libpthread-218.1.3 they now leave slots 6 and 11 unused to allow
+ * limited interoperability w/ code targeting the Windows x64 ABI. We steal slot 6
+ * for our own use.
```

Comment:
OK, one slot for access within compiled DR code is fine, but what is the plan for TLS from the code cache?  We can't indirect there: we need multiple directly addressable register scratch slots.

Reply:
As documented in #1568 option #3 seems to be the winner.